src/igraph/__init__.py

```diff
@@ -2850,6 +2850,84 @@ def Incidence(klass, *args, **kwds):
                     edge[weight_attr] = mat[target][source - num_vertices_of_first_kind]
         return result
 
+    @classmethod
+    def DataFrame(klass, edges, directed=True, vertices=None):
+        """DataFrame(directed=True, vertices=None)
+
+        Generates a graph from one or two dataframes.
+
+        @param edges: pandas DataFrame containing edges and metadata
+        @param directed: bool setting whether the graph is directed
+        @param vertices: None (default) or pandas DataFrame containing vertex
+          metadata. The first column must contain the unique ids of the
+          vertices and will be set as attribute 'name'. All other columns
+          will be added as vertex attributes by column name.
+
+        @return: the graph
+        """
+        import numpy as np
+        import pandas as pd
+
+        if edges.shape[1] < 2:
+            raise ValueError("the data frame should contain at least two columns")
+
+        # Handle if some elements are 'NA'
+        if edges.iloc[:, :2].isna().values.any():
+            warn("In `d' `NA' elements were replaced with string \"NA\"")
+            edges = edges.copy()
+            edges.iloc[:, :2].fillna('NA', inplace=True)
+
+        if (vertices is not None) and vertices.iloc[:, 0].isna().values.any():
+            warn("In `vertices.iloc[:, 0]` `NA' elements were replaced with string \"NA\"")
```

Comment:
I'm not that familiar with Pandas to judge this -- would "vertices.iloc[:, 0]" make sense for an average user that knows Pandas but has no idea about how this function is implemented internally in igraph? Can we come up with a more readable warning?

Reply:
fair enough, yes I put a more readable warning now